src/plugins/vis_builder/public/application/components/data_tab/field_selector_field.tsx

```diff
@@ -28,59 +28,119 @@
  * under the License.
  */
 
-import React, { useState } from 'react';
-import { IndexPatternField } from '../../../../../data/public';
-import { FieldButton, FieldIcon } from '../../../../../opensearch_dashboards_react/public';
-import { useDrag } from '../../utils/drag_drop/drag_drop_context';
-import { COUNT_FIELD } from '../../utils/drag_drop/types';
+import React, { useCallback, useState } from 'react';
+import { EuiPopover } from '@elastic/eui';
 
+import {
+  FilterManager,
+  IndexPattern,
+  IndexPatternField,
+  opensearchFilters,
+} from '../../../../../data/public';
+import {
+  FieldButton,
+  FieldButtonProps,
+  FieldIcon,
+} from '../../../../../opensearch_dashboards_react/public';
+
+import { COUNT_FIELD, useDrag } from '../../utils/drag_drop';
+import { VisBuilderFieldDetails } from './field_details';
+import { FieldDetails } from './types';
 import './field_selector_field.scss';
 
 export interface FieldSelectorFieldProps {
-  field: Partial<IndexPatternField> & Pick<IndexPatternField, 'displayName' | 'type' | 'scripted'>;
+  field: IndexPatternField;
+  filterManager: FilterManager;
+  indexPattern?: IndexPattern;
+  getDetails: (field) => FieldDetails;
 }
 
-// TODO:
-// 1. Add field sections (Available fields, popular fields from src/plugins/discover/public/application/components/sidebar/discover_sidebar.tsx)
-// 2. Add popover for fields stats from discover as well
-export const FieldSelectorField = ({ field }: FieldSelectorFieldProps) => {
+// TODO: Add field sections (Available fields, popular fields from src/plugins/discover/public/application/components/sidebar/discover_sidebar.tsx)
+export const FieldSelectorField = ({
+  field,
+  filterManager,
+  indexPattern,
+  getDetails,
+}: FieldSelectorFieldProps) => {
+  const { id: indexPatternId = '', metaFields = [] } = indexPattern ?? {};
+  const isMetaField = metaFields.includes(field.name);
```

Comment:
@ashwin-pc This is what discover does, but it's a bit different than the existing approach we had of hard-coding `META_FIELDS`. I'm not sure which is actually better. Is possible that different index patterns may actually have different `metaFields`?

Reply:
updated in latest revision to follow the discover pattern.